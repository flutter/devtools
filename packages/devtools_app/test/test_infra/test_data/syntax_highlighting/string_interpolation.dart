// Copyright 2022 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.

void values() {
  final i = 1;
  final j = 2;

  print('the value of \$i is $i');
  print('the value after \$i is ${i + 1}');
  print('the value of \$i + \$j is ${i + j}');
}

void functions() {
  print(
    '${() {
      return 'Hello';
    }}',
  );
  print(
    'print(${() {
      return 'Hello';
    }()})',
  );
  print('${() => 'Hello'}');
  print('print(${(() => 'Hello')()})');
}
