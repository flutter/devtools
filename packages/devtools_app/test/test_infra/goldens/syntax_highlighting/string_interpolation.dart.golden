>// Copyright 2022 The Chromium Authors. All rights reserved.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-slash.dart
>// Use of this source code is governed by a BSD-style license that can be
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-slash.dart
>// found in the LICENSE file.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-slash.dart
>
>void values() {
#^^^^ storage.type.primitive.dart
#     ^^^^^^ entity.name.function.dart
>  final i = 1;
#  ^^^^^ storage.modifier.dart
#          ^ keyword.operator.assignment.dart
#            ^ constant.numeric.dart
#             ^ punctuation.terminator.dart
>  final j = 2;
#  ^^^^^ storage.modifier.dart
#          ^ keyword.operator.assignment.dart
#            ^ constant.numeric.dart
#             ^ punctuation.terminator.dart
>
>  print('the value of \$i is $i');
#  ^^^^^ entity.name.function.dart
#        ^^^^^^^^^^^^^^ string.interpolated.single.dart
#                      ^^ string.interpolated.single.dart constant.character.escape.dart
#                        ^^^^^ string.interpolated.single.dart
#                             ^ string.interpolated.single.dart string.interpolated.expression.dart
#                              ^ string.interpolated.single.dart string.interpolated.expression.dart variable.parameter.dart
#                               ^ string.interpolated.single.dart
#                                 ^ punctuation.terminator.dart
>  print('the value after \$i is ${i + 1}');
#  ^^^^^ entity.name.function.dart
#        ^^^^^^^^^^^^^^^^^ string.interpolated.single.dart
#                         ^^ string.interpolated.single.dart constant.character.escape.dart
#                           ^^^^^ string.interpolated.single.dart
#                                ^^ string.interpolated.single.dart string.interpolated.expression.dart
#                                  ^ string.interpolated.single.dart string.interpolated.expression.dart variable.parameter.dart
#                                   ^^^ string.interpolated.single.dart string.interpolated.expression.dart
#                                      ^ string.interpolated.single.dart string.interpolated.expression.dart constant.numeric.dart
#                                       ^ string.interpolated.single.dart string.interpolated.expression.dart
#                                        ^ string.interpolated.single.dart
#                                          ^ punctuation.terminator.dart
>  print('the value of \$i + \$j is ${i + j}');
#  ^^^^^ entity.name.function.dart
#        ^^^^^^^^^^^^^^ string.interpolated.single.dart
#                      ^^ string.interpolated.single.dart constant.character.escape.dart
#                        ^^^^ string.interpolated.single.dart
#                            ^^ string.interpolated.single.dart constant.character.escape.dart
#                              ^^^^^ string.interpolated.single.dart
#                                   ^^ string.interpolated.single.dart string.interpolated.expression.dart
#                                     ^ string.interpolated.single.dart string.interpolated.expression.dart variable.parameter.dart
#                                      ^^^ string.interpolated.single.dart string.interpolated.expression.dart
#                                         ^ string.interpolated.single.dart string.interpolated.expression.dart variable.parameter.dart
#                                          ^ string.interpolated.single.dart string.interpolated.expression.dart
#                                           ^ string.interpolated.single.dart
#                                             ^ punctuation.terminator.dart
>}
>
>void functions() {
#^^^^ storage.type.primitive.dart
#     ^^^^^^^^^ entity.name.function.dart
>  print('${() {
#  ^^^^^ entity.name.function.dart
#        ^ string.interpolated.single.dart
#         ^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart
>    return 'Hello';
#^^^^ string.interpolated.single.dart string.interpolated.expression.dart
#    ^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart variable.parameter.dart
#          ^^^^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart
>  }}');
#^^^ string.interpolated.single.dart string.interpolated.expression.dart
#   ^^ string.interpolated.single.dart
#      ^ punctuation.terminator.dart
>  print('print(${() {
#  ^^^^^ entity.name.function.dart
#        ^^^^^^^ string.interpolated.single.dart
#               ^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart
>    return 'Hello';
#^^^^ string.interpolated.single.dart string.interpolated.expression.dart
#    ^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart variable.parameter.dart
#          ^^^^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart
>  }()})');
#^^^ string.interpolated.single.dart string.interpolated.expression.dart
#   ^^^^^ string.interpolated.single.dart
#         ^ punctuation.terminator.dart
>  print('${() => 'Hello'}');
#  ^^^^^ entity.name.function.dart
#        ^ string.interpolated.single.dart
#         ^^^^^^^^^^^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart
#                         ^ string.interpolated.single.dart
#                           ^ punctuation.terminator.dart
>  print('print(${(() => 'Hello')()})');
#  ^^^^^ entity.name.function.dart
#        ^^^^^^^ string.interpolated.single.dart
#               ^^^^^^^^^^^^^^^^^^^^ string.interpolated.single.dart string.interpolated.expression.dart
#                                   ^^ string.interpolated.single.dart
#                                      ^ punctuation.terminator.dart
>}
